views/index.ejs

```diff
@@ -8,9 +8,9 @@
     <h1>Shopping for Cookies</h1>
 
 
-    <strong>Items in cart: <!-- Print the number of items in the cart here --></strong>
+    <strong>Items in cart: <%= products.length %></strong>
     <ul>
-      <!-- List the names of the items in your cart here -->
+      <li><%= products.name %></li>
```

Comment:
You're on the right track with printing the name, but how do we loop through an array to print each product's name individually? You'd want to do a forEach loop and inside that loop, print product.name and product.id into a hidden input value so you can access the id later on. Take a look at the code below and see if this makes sense.

```js
<% cart.forEach(function(product){ %>
       <form action="'/cart'" method="post">
            <li><%= product.name %></li>
            <input type="hidden" name="_method" value="delete">
            <input type="hidden" name="id" value="<%= product.id %>">
      </form>
<% }); %>
```